src/Serializer/AbstractItemNormalizer.php

```diff
@@ -541,9 +541,14 @@ protected function denormalizeCollection(string $attribute, ApiProperty $propert
         $childContext = $this->createChildContext($this->createOperationContext($context, $className), $attribute, $format);
         $collectionKeyTypes = $type->getCollectionKeyTypes();
         foreach ($value as $index => $obj) {
+            $currentChildContext = $childContext;
+            if (isset($childContext['deserialization_path'])) {
+                $currentChildContext['deserialization_path'] = $childContext['deserialization_path'].'['.$index.']';
```

Comment:
```suggestion
                $currentChildContext['deserialization_path'] = "{$childContext['deserialization_path']}[{$index]}";
```

(micro-optim, but as we're on the hot path it may be worth it)

Reply:
Done.